### ISSUE TITLE
rawinput-deltatime fix

### ISSUE THREAD

**Issue by Scoovy** [nipkownix/re4_tweaks#262]
### Validation

- [X] Game has been updated to the latest RE4HD release (v1.0.1: https://www.re4hd.com/?p=9471)
- [X] Data has been [validated with SFV](https://www.re4hd.com/?p=9454) (a guide to using QuickSFV is available at that link, use that with this 1.0.1 SFV: [BIO4-HDProject1.0.1-SFV.zip](https://github.com/nipkownix/re4_tweaks/files/8057899/BIO4-HDProject1.0.1-SFV.zip))

### Describe your issue here (drag+drop ZIP to attach it)

in the new updates of the DLL there was a specific update that has an option that cant be changed that makes aiming super fast,

This Version added it and all others after
![unknown](https://user-images.githubusercontent.com/103059961/178117924-c7576aa9-002e-4936-b758-f4c318cee6ee.png)

Please remove or make the Deltatime version a toggle as versions without this had better aiming speed

**Comment by emoose** [comment]
You could try setting mouse sensitivity option to half of its current value, think that should give you the same speed as it was before.

**Comment by nipkownix** [comment]
Yeah, as emoose said, you should just lower the game's sensitivity. Unless you already have done that and it is still too sensitive?

**Comment by Scoovy** [comment]
> Yeah, as emoose said, you should just lower the game's sensitivity. Unless you already have done that and it is still too sensitive?

yeah the problem with that is the camera turning speed when not aiming does not have the same speed as aiming so when i turn the sensitivity down to a point where its good the camera will be way slower than it should be

**Comment by Scoovy** [comment]
> Hmm, then we might be missing the same delta adjustment for the camera as well? @emoose any chance that's the case?
> 
> Also, @Scoovy, you mean the camera without mouse turning enabled, right?

yes without mouse turning

**Comment by emoose** [comment]
>then we might be missing the same delta adjustment for the camera as well? @emoose any chance that's the case?

Yeah sounds like that's probably it, not really sure why though, I thought the MouseDelta / UseRawMouseInput hooks would be changing the stuff for that already, but I guess maybe that's only affecting aiming, and the normal camera is being updated some other way that we haven't hooked yet? hm

**Comment by nipkownix** [comment]
>but I guess maybe that's only affecting aiming, and the normal camera is being updated some other way that we haven't hooked yet?

Yeah, that's probably it. I never noticed it because, ever since MouseTurning became a thing, I never played without it. Hm.

**Comment by emoose** [comment]
Might have a fix for it, just adjusted the MouseDelta stuff so it'll use the slower 60FPS speed instead of basing it on the faster 30FPS one.

@Scoovy could you try this build out and see if it's any better? You should be able to put mouse sensitivity back to how you had it before, so you can have camera speed back to normal:
[dinput8-mouseDeltaTimeFix.zip](https://github.com/nipkownix/re4_tweaks/files/9078209/dinput8-mouseDeltaTimeFix.zip)

**Comment by DarkxVoid-loop** [comment]
@emoose I tested the fix and it still has the camera of the sights faster, you couldn't leave this feature of you guys as optional because it made it unplayable and before it was better

**Comment by nipkownix** [comment]
Not sure what you mean by "camera of the sights".

**Comment by DarkxVoid-loop** [comment]
Leon's camera is slow in relation to the weapon's camera, this is making it unplayable, if you change the sensitivity of one it makes the other worse

**Comment by FutonGama** [comment]
Ok, Keep It with the "60FPS speed instead of always using the crazy fast 30FPS speed."

**Comment by nipkownix** [comment]
Is that version good then?
(I haven't had any time to test re4 stuff lately, sorry)

**Comment by emoose** [comment]
Hmm, that mouseDeltaTimeFix should have made it like it was before (unless you're playing at 30FPS, which would now match the 60FPS speed, instead of the reverse like the first fix was)

eg. the original code worked like
>delta = movement * sensitivity

so say sensitivity is 15, movement 3, the amount of movement/delta would be 45

The original mouseDelta change edited that so that frametime is used to make it consistent across different framerates, eg:
>delta = (movement * sensitivity) / frametime

with frametime being either 0.5 (60fps) or 1 (30fps), so if we say that framerate is 60, with the same values above this gives us delta of 90 (explaining why aiming speed increased so much)

The final mouseDeltaTimeFix I posted would just divide the `(movement * sensitivity)` by 2 before dividing by frametime (pretty much the same as halving the mouse sensitivity option, except this way won't affect camera speed)

If we run the numbers with that, we get a delta of 45, the original value before any of the deltaTime stuff was added.

The advantage we get is that running at 30FPS should now also get the same result as running at 60FPS too, and same when running at 90/120/1000+FPS.

(sorry for the math lesson, just wanted to double check it myself, and seemed like best way of explaining it too :)

Anyway for me, with the mouseDeltaTimeFix the aiming & camera speed seems to be the same as the 1.7.6 release before any of the deltaTime changes, without needing to change mouse sensitivity, the only difference should be when playing at non-60FPS framerates.

If sensitivity is still different for you maybe try resetting usr_input.ini and then compare with 1.7.6, hopefully it should match up with it.
